app/src/main/java/com/deitel/tipcalculator/MainActivity.java

```diff
@@ -2,116 +2,177 @@
-            boolean fromUser) {
-            percent = progress / 100.0; // set percent based on progress
-            calculate(); // calculate and display tip and total
-         }
-
-         @Override
-         public void onStartTrackingTouch(SeekBar seekBar) { }
-
-         @Override
-         public void onStopTrackingTouch(SeekBar seekBar) { }
-      };
-
-   // listener object for the EditText's text-changed events
-   private final TextWatcher amountEditTextWatcher = new TextWatcher() {
-      // called when the user modifies the bill amount
-      @Override
-      public void onTextChanged(CharSequence s, int start,
-         int before, int count) {
-
-         try { // get bill amount and display currency formatted value
-            billAmount = Double.parseDouble(s.toString()) / 100.0;
-            amountTextView.setText(currencyFormat.format(billAmount));
-         }
-         catch (NumberFormatException e) { // if s is empty or non-numeric
-            amountTextView.setText("");
-            billAmount = 0.0;
-         }
-
-         calculate(); // update the tip and total TextViews
-      }
-
-      @Override
-      public void afterTextChanged(Editable s) { }
-
-      @Override
-      public void beforeTextChanged(
-         CharSequence s, int start, int count, int after) { }
-   };
+    // currency and percent formatter objects
+    private static final NumberFormat currencyFormat =
+            NumberFormat.getCurrencyInstance();
+    private static final NumberFormat percentFormat =
+            NumberFormat.getPercentInstance();
+
+    private double billAmount = 0.0; // bill amount entered by the user
+    private double percent = 0.15; // initial tip percentage
+    private TextView amountTextView; // shows formatted bill amount
+    private TextView percentTextView; // shows tip percentage
+    private TextView tipTextView; // shows calculated tip amount
+    private TextView totalTextView; // shows calculated total bill amount
+    private double peopleAmount = 0;    // zin
+    private TextView peopleTextView;    // zin
+    private TextView eachTextView;      //zin
+
+
+    // called when the activity is first created
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState); // call superclass onCreate
+        setContentView(R.layout.activity_main); // inflate the GUI
+
+        // get references to programmatically manipulated TextViews
+        amountTextView = (TextView) findViewById(R.id.amountTextView);
+        percentTextView = (TextView) findViewById(R.id.percentTextView);
+        tipTextView = (TextView) findViewById(R.id.tipTextView);
+        totalTextView = (TextView) findViewById(R.id.totalTextView);
+        tipTextView.setText(currencyFormat.format(0));
+        totalTextView.setText(currencyFormat.format(0));
+        peopleTextView = (TextView) findViewById(R.id.peopleTextView);  // zin
```

Comment:
Since you never change the text in _peopleTextView_, there is no reason to get a reference to it.
Remember, the data goes from _peopleEditText_ to _eachTextView_